src/soc/ibm/power9/chip.c

```diff
@@ -512,16 +513,22 @@ static void add_tpm_node(struct device_tree *tree)
 	snprintf(compatible, sizeof(compatible), "infineon,%s", tis_name());
 
 	dt_add_string_prop(tpm, "compatible", strdup(compatible));
-	dt_add_string_prop(tpm, "status", "okay");
 	dt_add_u32_prop(tpm, "reg", addr);
 
 	tcpa_table = cbmem_find(CBMEM_ID_TCPA_LOG);
 	if (tcpa_table == NULL)
 		die("TPM events (TCPA) log is missing from CBMEM!");
 
-	dt_add_u64_prop(tpm, "linux,sml-base", (uintptr_t)tcpa_table + sizeof(*tcpa_table));
+	dt_add_u64_prop(tpm, "linux,sml-base", (uintptr_t)&tcpa_table->header);
 	dt_add_u32_prop(tpm, "linux,sml-size",
+			sizeof(struct tcpa_header) +
 			tcpa_table->max_entries * sizeof(struct tcpa_entry));
+
+	/* Not hard-coding into DTS-file in case will need to store key hash here */
+	sb = dt_find_node_by_path(tree, "/ibm,secureboot", NULL, NULL, 1);
+	dt_add_string_prop(sb, "compatible", "ibm,secureboot-v1-softrom");
```

Comment:
Depending on securerom will make bootblock in SEEPROM difficult (HBBL has securerom appended which takes almost 9KB out of 20KB available), but we will worry about it when we get there.

Reply:
The only consideration behind picking `softrom` was to involve less initialization code and get to TPM usage faster.

Reply:
Makes sense, the code is compiled in anyway.